packages/compass-editor/src/ace/query-autocompleter.ts

```diff
@@ -0,0 +1,49 @@
+import type { Ace } from 'ace-builds';
+import type { CompletionWithServerInfo } from '../types';
+import { filter, MATCH_COMPLETIONS } from './util';
+
+/**
+ * Adds autocomplete suggestions for queries.
+ */
+class QueryAutoCompleter implements Ace.Completer {
+  constructor(
+    public version: string,
+    public textCompleter: Ace.Completer,
```

Comment:
I'm planning to remove this from all constructors now that it all in the same package, but will keep it for next PR to not introduce any interface changes in this PR